File: src/org/usfirst/frc5293/Subsystems.java
package org.usfirst.frc5293;

import org.usfirst.frc5293.subsystems.*;

public class Subsystems {
    private static Drivetrain drivetrain;
    private static Camera camera;
    private static CameraRingLight cameraRingLight;
    private static Shooter shooter;

    public static void init() {
        drivetrain = new Drivetrain();
        camera = new Camera();
        cameraRingLight = new CameraRingLight();
        shooter = new Shooter();
    }

    public static Drivetrain getDrivetrain() {
        return drivetrain;
    }

    public static Camera getCamera() {
        return camera;
    }

    public static CameraRingLight getCameraRingLight() { return cameraRingLight; }

    public static Shooter getShooter() {
        return shooter;
    }
}

File: src/org/usfirst/frc5293/Subsystems.kt
package org.usfirst.frc5293

import org.usfirst.frc5293.subsystems.*
import org.usfirst.frc5293.util.Initializable
import org.usfirst.frc5293.util.LazyGroup

object Subsystems : Initializable {

    lateinit var drivetrain: Drivetrain
    lateinit var camera: Camera
    lateinit var cameraRingLight: CameraRingLight
    lateinit var shooter: Shooter
    lateinit var shooterKicker: ShooterKicker

    override fun init() {
        drivetrain = Drivetrain()
        camera = Camera()
        cameraRingLight = CameraRingLight()
        shooter = Shooter()
        shooterKicker = ShooterKicker()
    }
}
